sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/StateStore.scala

```diff
@@ -36,10 +36,14 @@ import org.apache.spark.sql.types.StructType
 import org.apache.spark.util.{ThreadUtils, Utils}
 
 /**
- * Base trait for a versioned key-value store. Each instance of a `StateStore` represents a specific
- * version of state data, and such instances are created through a [[StateStoreProvider]].
+ * Base trait for a versioned key-value store which provides read operations. Each instance of a
+ * `ReadStateStore` represents a specific version of state data, and such instances are created
+ * through a [[StateStoreProvider]].
+ *
+ * `abort` method will be called when the task is completed - please clean up the resources in
+ * the method.
  */
-trait StateStore {
+trait ReadStateStore {
```

Comment:
How about renaming this to ReadOnlyStateStore. I think it should be a better name consistent with other new functions in this PR, e.g mapPartitionsWithReadOnlyStateStore, ReadOnlyStateStoreRDD, etc

Reply:
+1 on this, I've suggested the same but there was another agreement before.

Reply:
Okay for me if you both think it is better. Not strong opinion here.

Reply:
It's your call then @HeartSaVioR :)

Reply:
OK I'll rename it. No big deal.

Reply:
Ah sorry I reminded the comment from @viirya and still think his comment makes more sense. I'll rename others to make consistent for now, but I'm also open to rename this to ReadOnlyStateStore, if both of you think it's better after reading the comment from @viirya .

@xuanyuanking @gaborgsomogyi Could you please go through the previous comment below and comment how you think?
https://github.com/apache/spark/pull/26935#discussion_r491747706

Reply:
Ah sorry +1... Didn't notice the previous discussion.
I think to rename others for consistency is enough.

Reply:
Just gone through and makes sense, +1.